src/python/pants/backend/python/framework/stevedore/target_types.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import collections.abc
+import os
+from dataclasses import dataclass
+
+from pants.backend.python.target_types import EntryPoint, PythonResolveField
+from pants.engine.addresses import Address
+from pants.engine.collection import Collection
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    AsyncFieldMixin,
+    Dependencies,
+    Field,
+    InvalidFieldException,
+    InvalidFieldTypeException,
+    SecondaryOwnerMixin,
+    StringField,
+    StringSequenceField,
+    Target,
+    Targets,
+)
+from pants.source.filespec import Filespec
+from pants.util.strutil import softwrap
+
+
+@dataclass(frozen=True)
+class StevedoreEntryPoint:
+    name: str
+    value: EntryPoint
+
+
+class StevedoreEntryPoints(Collection[StevedoreEntryPoint]):
+    pass
+
+
+class StevedoreNamespaceField(StringField):
+    alias = "namespace"
+    help = softwrap(
+        """
+        Set the stevedore extension namespace.
+
+        This looks like a python module 'my.stevedore.namespace', but a python module
+        of that name does not need to exist. This is what a stevedore ExtensionManager
+        uses to look up relevant entry_points from pkg_resources.
+        """
+    )
+    required = True
+
+
+class StevedoreEntryPointsField(AsyncFieldMixin, SecondaryOwnerMixin, Field):
+    # based on pants.backend.python.target_types.PexEntryPointField
+    # and on pants.backend.python.engine.target.DictStringToStringField
```

Comment:
I'm not sure if we want these origin comments in the Pants repo..? (for st2 I can see it makes sense to have them)